File: medium/strings/longestPalindrome.py
def longestPalindrome(s):
	n = len(s)
	low = 0; high = 0; start = 0; maxlength = 1;

	for x in range(1, n):
		# for even length substring
		low = x-1
		high = x
		while low >= 0 and high < n and s[low] == s[high]:
			if maxlength < high-low+1:
				maxlength = high-low+1
				start = low
			low -= 1
			high += 1


		# for odd length substring
		low = x-1
		high = x+1
		while low >= 0 and high < n and s[low] == s[high]:
			if maxlength < high-low+1:
				maxlength = high-low+1
				start = low
			low -= 1
			high += 1	

	return s[start:start+maxlength]	
		

File: medium/strings/longestPalindrome.cpp


string longestPalindrome(string s){
	int n = s.size();
	int low, high, maxlength = 1, start = 0;
	if(n == 1)
		return s;
	for(int i = 1; i < n; ++i){
		// for even legth substring
		low = i-1;
		high = i;

		while(low >= 0 && high < n && (s[low] == s[high])){
			if(maxlength < high-low+1){
				maxlength = high-low+1;
				start = low; // starting index of longest palindrome
			}
			// increasing length of substring
			low--;
			high++;
		}

		// for odd substring
		low = i-1;
		high = i+1;

		while(low >= 0 && high < n && (s[low] == s[high])){
			if(maxlength < high-low+1){
				maxlength = high-low+1;
				start = low;
			}
			low--;
			high++;
		}
	}    

	// cout << maxlength << endl;
	string ans = s.substr(start, maxlength);
	return ans;    
}

